reactor-core/src/main/java/reactor/util/context/Context.java

```diff
@@ -89,6 +116,13 @@ static Context of(Object key1, Object value1,
 	static Context of(Object key1, Object value1,
 			Object key2, Object value2,
 			Object key3, Object value3) {
+		//prefer direct check for the simple 3 keys case over checkDuplicateKeys (saves up 1 array allocation)
+		if (key1.equals(key2) || key1.equals(key3)) {
```

Comment:
there is no null check so it is possible to catch NPE.  
Intresting. ContextX constructors allow null keys and values but put() method does not

Reply:
mmh that is indeed a shortcoming, although the IDE should report this kind of usage (as part of a `@NonNullApi`). let me open a separate issue to enforce null checks in the constructors.

Reply:
see #1797 and PR #1800